cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/constraint/AbstractConstraint.java

```diff
@@ -24,13 +24,15 @@
 /**
  * A constraint that is checked before executing a schedule.
  */
-public class ConstraintChecker {
+public abstract class AbstractConstraint implements Constraint {
```

Comment:
I think it's better to have an interface `CheckableConstraint` or `JobConstraint` that is internal to the scheduler in app-fabric and have the abstract class implementing that interface.